recipe-client-addon/bootstrap.js

```diff
@@ -29,69 +38,147 @@ const DEFAULT_PREFS = {
   "app.shield.optoutstudies.enabled": AppConstants.MOZ_DATA_REPORTING,
 };
 
-this.install = function() {};
-
-this.startup = function() {
-  // Initialize preference defaults before anything else happens.
-  const prefBranch = Services.prefs.getDefaultBranch("");
-  for (const [name, value] of Object.entries(DEFAULT_PREFS)) {
-    switch (typeof value) {
-      case "string":
-        prefBranch.setCharPref(name, value);
-        break;
-      case "number":
-        prefBranch.setIntPref(name, value);
-        break;
-      case "boolean":
-        prefBranch.setBoolPref(name, value);
-        break;
-      default:
-        throw new Error(`Invalid default preference type ${typeof value}`);
+// Logging
+const log = Log.repository.getLogger(BOOTSTRAP_LOGGER_NAME);
+log.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
+log.level = Services.prefs.getIntPref(PREF_LOGGING_LEVEL, Log.Level.Warn);
+
+this.Bootstrap = {
+  initShieldPrefs(defaultPrefs) {
+    const prefBranch = Services.prefs.getDefaultBranch("");
+    for (const [name, value] of Object.entries(defaultPrefs)) {
+      switch (typeof value) {
+        case "string":
+          prefBranch.setCharPref(name, value);
+          break;
+        case "number":
+          prefBranch.setIntPref(name, value);
+          break;
+        case "boolean":
+          prefBranch.setBoolPref(name, value);
+          break;
+        default:
+          throw new Error(`Invalid default preference type ${typeof value}`);
+      }
     }
-  }
+  },
 
-  ShieldRecipeClient.startup();
-};
+  initExperimentPrefs() {
+    const defaultBranch = Services.prefs.getDefaultBranch("");
+    const experimentBranch = Services.prefs.getBranch(STARTUP_EXPERIMENT_PREFS_BRANCH);
+
+    for (const childPrefName of experimentBranch.getChildList("")) {
+      const experimentPrefType = experimentBranch.getPrefType(childPrefName);
+      const realPrefName = childPrefName.slice(1); // Remove leading dot
```

Comment:
Would it make sense to put a dot at the end of `STARTUP_EXPERIMENT_PREFS_BRANCH`, so that this isn't needed?